src/core/syscalls/syscall_request.rs

```diff
@@ -6,9 +6,18 @@ use cairo_rs::{types::relocatable::Relocatable, vm::vm_core::VirtualMachine};
 
 #[derive(Debug, PartialEq)]
 pub(crate) enum SyscallRequest {
+    EmitEvent(EmitEventStruct),
     SendMessageToL1(SendMessageToL1SysCall),
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct EmitEventStruct {
```

Comment:
```suggestion
pub(crate) struct EmitEventRequest {
```
or 
```suggestion
pub(crate) struct EmitEventSysCall {
```

Just to keep some kind of convention

Reply:
In general, let's consider calling a `struct` `Struct` to be redundant and a little smelly. We already know it's a `struct` because it has fields.